File: Problems/Mine approach/Bit Manipulation/js/Single Number.js
// Given an array of integers A, every element appears twice except for one. Find that single one
module.exports = { 
    //param A : array of integers
    //return an integer
       singleNumber : function(A){
        let n = 0;
        for(let i=0; i<A.length; i++){
            n = n^A[i];
        }
        return n;
	}
};

File: Problems/Mine approach/Bit Manipulation/cpp/Single number.cpp
// Given an array of integers A, every element appears twice except for one. Find that single one
int Solution::singleNumber(const vector < int > & A) {
    int n = A.size();
    int result = 0;
    for (int i = 0; i < n; i++) {
        result ^= A[i];
    }
    return result;
}